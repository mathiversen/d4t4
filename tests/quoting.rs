use data_parser::{parse, Result};
use indoc::indoc;
use insta::assert_json_snapshot;

#[test]
fn it_can_parse_double_quotes() -> Result<()> {
    let markup = indoc!(
        r#"{
            "name": "Mr. Anderson",
            "greeting": "We meet again &{name}",
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_can_parse_single_quotes() -> Result<()> {
    let markup = indoc!(
        r#"{
            'name': 'Mr. Anderson',
            'greeting': 'We meet again &{name}',
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_can_parse_mixed_quotations() -> Result<()> {
    let markup = indoc!(
        r#"{
            "name": 'Mr. Anderson',
            "greeting": 'We "meet" again &{name}',
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_can_parse_double_inside_single_quotes() -> Result<()> {
    let markup = indoc!(
        r#"{
            'name': 'Mr. Anderson',
            'greeting': 'We "meet" again &{name}',
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_can_parse_single_inside_double_quotes() -> Result<()> {
    let markup = indoc!(
        r#"{
            "name": "Mr. Anderson",
            "greeting": "We 'meet' again &{name}",
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_can_parse_double_with_escaped_double() -> Result<()> {
    let markup = indoc!(
        r#"{
            "name": "Mr. Anderson",
            "greeting": "We \"meet\" again &{name}",
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_can_parse_single_with_escaped_single() -> Result<()> {
    let markup = indoc!(
        r#"{
            'name': 'Mr. Anderson',
            'greeting': 'We \'meet\' again &{name}',
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}
