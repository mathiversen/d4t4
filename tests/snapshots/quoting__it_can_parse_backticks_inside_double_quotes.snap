---
source: tests/quoting.rs
expression: x
---
{
  "greeting": "We `meet` again Mr. Anderson",
  "name": "Mr. Anderson"
}
